.github/workflows/node.js.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [main, testing]
+  pull_request:
+    branches: [main, testing]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+
+      - run: yarn install --frozen-lockfile
```

Comment:
The --frozen-lockfile option is deprecated; use --immutable and/or --immutable-cache instead

Reply:
> The --frozen-lockfile option is deprecated; use --immutable and/or --immutable-cache instead

https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

Documentation I refer to 🤔

Reply:
> > The --frozen-lockfile option is deprecated; use --immutable and/or --immutable-cache instead
> 
> [https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md?rgh-link-date=2022-05-19T11%3A57%3A28Z#caching-packages-data)
> 
> Documentation I refer to 🤔

This doc for yarn v1